ddprof-lib/src/main/cpp/flightRecorder.cpp

```diff
@@ -123,16 +123,15 @@ void Lookup::cutArguments(char* func) {
     }
 }
 
-void Lookup::fillJavaMethodInfo(MethodInfo* mi, jmethodID method, bool first_time) {
-    JNIEnv* jni = VM::jni();
+void Lookup::fillJavaMethodInfo(JNIEnv* jni, MethodInfo* mi, jmethodID method, bool first_time) {
     if (jni->PushLocalFrame(64) != 0) {
```

Comment:
Do we really need this patch? We theoretically have the protection of this local frame which will get popped very quickly. Do we fail with the reproducer program from upstream?

Reply:
The reproducer has nothing to do with the profiler. It is only JVMTI code so I expect it will fail if I run it on my machine.
And the patch is lowering the likelihood of the crash with [JDK-8268364 ](https://bugs.openjdk.org/browse/JDK-8268364). Without that fix this patch is mostly a no-op, just being a nicer citizen and releasing local refs promptly (they will get released when the JNI call returns to JVM - but in this case it means once the whole recording is dumped).

Reply:
But because there's a local frame here, the reference will be will get released when the frame is [popped](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#PopLocalFrame) on method exit anyway. I.e. I think we're already working around this.